tests/tally/generator.rs

```diff
@@ -181,55 +191,51 @@ impl VoteRoundGenerator {
                             .iter()
                             .find(|(id, _v)| **id == self.voting_token)
                             .map(|(_, v)| *v)
-                            .unwrap()
+                            .unwrap_or(Value(0))
                     })
                     .sum(),
             )
-            .unwrap();
+            .unwrap()
+    }
 
-        TokenDistribution::new(token_totals, ledger.accounts().clone())
+    pub fn ledger(&self) -> Ledger {
+        let block0 = self.block0.to_block();
+        Ledger::new(block0.header().id(), block0.fragments()).unwrap()
     }
 
     /// Tally voteplans and return the fragments to run the tally in a separate ledger
     pub fn tally_transactions<R: Rng + CryptoRng>(&mut self, rng: &mut R) -> Vec<Fragment> {
         let mut fragments = Vec::new();
-        let block0 = self.block0.to_block();
         let member_keys = self
             .committee_manager
             .members()
             .iter()
             .map(|member| member.public_key())
             .collect::<Vec<_>>();
 
-        let tmp_ledger = Ledger::new(block0.header().id(), block0.fragments()).unwrap();
+        let tmp_ledger = self.ledger();
         let stake_control = StakeControl::new_with(tmp_ledger.accounts(), &utxo::Ledger::new());
-        let table = chain_vote::TallyOptimizationTable::generate(stake_control.assigned().into());
+        let table = chain_vote::TallyOptimizationTable::generate(
+            NonZeroU64::new(stake_control.assigned().into()).unwrap(),
```

Comment:
I think this is probably not correct, but I'm not sure exactly what is the right thing to do. I imagine we may want to use `batch_decrypt` here too.

This may work nonetheless if the native stake is bigger than the tokens though.

Reply:
I tried to change to batch_decrypt, but keep recieving ` CannotTallyVotes { source: InvalidDecryption }'` in the test. I think this change exceeds my knowledge